test/games/strategy/engine/data/UnitCollectionTest.java

```diff
@@ -110,252 +94,292 @@ public void addUnit() {
     final int expNotifyChangedCountAfter = notifyChangedCountBefore + 1;
     assertThat(defaultPlayerID.getNotifyChangedCounter(), is(equalTo(expNotifyChangedCountAfter)));
   }
 
   @Test
   public void getUnitCount() {
-    assertThat(uc.getUnitCount(), is(equalTo(0)));
-    uc.addAllUnits(getUnitsOfDefaultPlayerOfUnitTypeOne());
-    assertThat(uc.getUnitCount(), is(equalTo(unitCountDefaultPlayerUnitTypeOne)));
-    uc.addAllUnits(getUnitsOfDefaultPlayerOfUnitTypeTwo());
+    assertThat(unitCollection.getUnitCount(), is(equalTo(0)));
+    addAllUnitTypeOneUnitsOfDefaultPlayerToUnitCollection();
+    assertThat(unitCollection.getUnitCount(), is(equalTo(unitCountDefaultPlayerUnitTypeOne)));
+    addAllUnitTypeTwoUnitsOfDefaultPlayerToUnitCollection();
     final int expUnitCount = unitCountDefaultPlayerUnitTypeOne + unitCountDefaultPlayerUnitTypeTwo;
-    assertThat(uc.getUnitCount(), is(equalTo(expUnitCount)));
+    assertThat(unitCollection.getUnitCount(), is(equalTo(expUnitCount)));
+  }
+
+  private void addAllUnitTypeTwoUnitsOfDefaultPlayerToUnitCollection() {
+    unitCollection.addAllUnits(getUnitsOfDefaultPlayerOfUnitTypeTwo());
+  }
+
+  private void addAllUnitTypeOneUnitsOfDefaultPlayerToUnitCollection() {
+    unitCollection.addAllUnits(getUnitsOfDefaultPlayerOfUnitTypeOne());
   }
 
   @Test
   public void getUnitCountByUnitType() {
-    assertThat(uc.getUnitCount(unitTypeOne), is(equalTo(0)));
-    assertThat(uc.getUnitCount(unitTypeTwo), is(equalTo(0)));
+    assertThat(unitCollection.getUnitCount(unitTypeOne), is(equalTo(0)));
+    assertThat(unitCollection.getUnitCount(unitTypeTwo), is(equalTo(0)));
 
-    addAllDefaultPlayerUnits();
-    assertThat(uc.getUnitCount(unitTypeOne), is(equalTo(unitCountDefaultPlayerUnitTypeOne)));
-    assertThat(uc.getUnitCount(unitTypeTwo), is(equalTo(unitCountDefaultPlayerUnitTypeTwo)));
+    addAllDefaultPlayerUnitsToUnitCollection();
+    assertThat(unitCollection.getUnitCount(unitTypeOne), is(equalTo(unitCountDefaultPlayerUnitTypeOne)));
+    assertThat(unitCollection.getUnitCount(unitTypeTwo), is(equalTo(unitCountDefaultPlayerUnitTypeTwo)));
 
-    addAllOtherPlayerUnits();
-    assertThat(uc.getUnitCount(unitTypeOne), is(equalTo(unitCountUnitTypeOne)));
-    assertThat(uc.getUnitCount(unitTypeTwo), is(equalTo(unitCountUnitTypeTwo)));
+    addAllOtherPlayerUnitsToUnitCollection();
+    assertThat(unitCollection.getUnitCount(unitTypeOne), is(equalTo(unitCountUnitTypeOne)));
+    assertThat(unitCollection.getUnitCount(unitTypeTwo), is(equalTo(unitCountUnitTypeTwo)));
+  }
+
+  private void addAllOtherPlayerUnitsToUnitCollection() {
+    unitCollection.addAllUnits(getUnitsOfOtherPlayerOfUnitTypeOne());
+    unitCollection.addAllUnits(getUnitsOfOtherPlayerOfUnitTypeTwo());
+  }
+
+  private Collection<Unit> getUnitsOfOtherPlayerOfUnitTypeTwo() {
+    final Collection<Unit> units = new ArrayList<Unit>();
+    units.add(unitOtherPlayer4);

```

Comment:
I would just inline 'unitOtherPlayer4' here. It looks like it is only used in just this one spot.  That would reduce a variable from the class level scope and tuck it into the method (the goal is information hiding)